contentcuration/search/tests/test_search.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import absolute_import
+
+from django.urls import reverse
+
+from contentcuration.models import Channel
+from contentcuration.tests import testdata
+from contentcuration.tests.base import StudioAPITestCase
+
+
+class SearchViewsetTestCase(StudioAPITestCase):
+
+    def test_filter_exclude_channels(self):
+        user = testdata.user()
+        self.client.force_authenticate(user=user)
+        channel = testdata.channel()
+        channel.editors.add(user)
+        response = self.client.get(
+            reverse("search-list"), data={"exclude_channel": channel.id}, format="json",
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+        self.assertEqual(response.data["results"], [])
+
+    def test_filter_channels_by_edit(self):
+        user = testdata.user()
+        self.client.force_authenticate(user=user)
+        channel = testdata.channel()
+        channel.editors.add(user)
+        response = self.client.get(
+            reverse("search-list"), data={"channel_list": "edit"}, format="json",
+        )
+        self.assertEqual(response.status_code, 200, response.content)
+        self.assertNotEqual(response.data["results"], [])
+
+    def test_search(self):
+        users = []
+        channels = []
+
+        for i in range(4):
+            user = testdata.user(email="a{}@a.com".format(i))
+            users.append(user)
+
+            channel = Channel.objects.create(name="user_a{}_channel".format(i))
+            channel.save()
+            channels.append(channel)
+            channel.editors.add(user)
+
+        public_channel, editable_channel, viewable_channel, inaccessible_channel = channels
```

Comment:
This seems to be covering all the different options, and the assertions seem correct!

Reply:
This test was written originally by @bjester for my changes to have no regression. Thanks and credits to him sir! :smile: